client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HotRodIntegrationTest.java

```diff
@@ -161,6 +163,22 @@ public void testReplaceIfUnmodified() {
       assert !remoteCache.replaceWithVersion("aKey", "aNewValue", valueBinary.getVersion());
    }
 
+   public void testPutAllAndReplaceWithVersion(Method m ) {
```

Comment:
probably worth adding another test with `putAll()` and `removeWithVersion()`

Reply:
and checking that values `A` and `B` have different versions. 
```java
remoteCache.putAll(Map.of(key, "A"))
remoteCache.putAll(Map.of(key, "B"))
```

Reply:
Added test covering these cases. Thanks!